### PR TITLE
feat: Add Issue Mod Flows

### LINKED ISSUE #446: [Advanced]: Add Workflow to Triage Issues
### 🧑‍🔬 Advanced Issue

Welcome! This is an **[Advanced Issue](https://github.com/issues?q=is%3Aopen%20is%3Aissue%20org%3Ahiero-ledger%20archived%3Afalse%20no%3Aassignee%20label%3A%22advanced%22%20repo%3Ahiero-ledger%2Fhiero-website)** touching core website architecture.

It is designed for expert contributors who have demonstrated deep architectural understanding and a proven track record of high-quality contributions.

### 🐞 Problem Description

Currently when a contributor creates an issue no one is notified and the issue remains open. 

This creates a problem as the team needs to determine if the issue is valid and label it properly.
With out proper labeling our current bots to do not count the work. This interferes with issue counts for contributors. Ultimately, the bot does not recognize the issue or add it to the count with proper triage.  

### 🛠️ Implementation Notes

We propose to implement work flow files that will moderate issues by reserving/locking them until the issue has been approved by a team member. 

Workflows must be tested and corresponding PR's must include some form of successful testing to ensure code works as intended. 

### 🧠 Advanced Contributors — Prerequisites & Expectations

> [!CAUTION]
> **Advanced issues are the highest-risk work in this project. We will reject PRs that do not meet these standards.**

### 🏁 Concrete Prerequisites
- **Advanced Web Development:** Proficient with React, Next.js, and TypeScript.
- **Expertise:** Deep architectural understanding of the website's component structure, routing, and build pipeline.
- **Proven History:** Successfully completed **≥ 10 intermediate issues** in this repo.
- **Consistency:** **≥ 3–4 months** of active, human-led contributions to this project.

> [!NOTE]
> **CI/CD Exception:** For issues focused on **GitHub Actions / Workflows**, the repo-specific thresholds above may be waived if the contributor demonstrates advanced-level proficiency in CI/CD.

### ⚠️ AI Usage Policy

- Using AI to generate code for Advanced issues is **strictly discouraged**
- AI may be used to help explain file relationships, but cannot be the main source of research.
- Submitting AI-generated or unvalidated code is grounds for **immediate closure**

### ⏱️ Timeline & Workflow
- **Typical time:** ~1 month / ~50 hours.
- 🔴 Completing an advanced issue in 1–3 days is a **red flag** and will likely be rejected.
- **Suggested:** Post your proposed architectural approach as a comment and wait for explicit maintainer approval **before writing any code.**

### 🔬 Technical Domains

- [ ] **Advanced React / Next.js** (Server Components, App Router, dynamic routing, middleware)
- [ ] **Component Architecture** (design systems, reusable patterns, composition)
- [ ] **Performance & Optimization** (Core Web Vitals, image optimization, code splitting, caching)
- [ ] **Accessibility** (WCAG compliance, screen reader support, keyboard navigation)
- [ ] **Build & Deployment** (Next.js config, CI/CD, hosting/deployment integration)
- [ ] **Testing** (unit, integration, E2E, accessibility testing, visual regression)

### 🛡️ Quality & Review Standards

Advanced PRs must be **"safe, maintainable, architecturally sound, and production-ready."**

1. **Architectural Fit:** The solution must fit naturally into the existing website architecture.
2. **Security & Correctness:** Evaluate all logic for injection risks, XSS vulnerabilities, or state corruption issues. 
3. **Maintainability:** Code must be short and clear enough for any other maintainer to debug without your assistance. 
4. **Backward Compatibility:** Existing pages, routes, and components must not break. If a breaking change is required, it must be explicitly managed.
5. **Comprehensive Testing:** Must include unit and integration tests covering all new logic paths, edge cases, and failure modes. AI-generated tests based on AI-generated code are grounds for immediate rejection.

### ✅ PR Quality Checklist

Before opening your PR, the contributor must confirm:
- [ ] I have spent the majority of my time researching the problem and solution space extensively, including reviewing relevant code, documentation, and external resources.
- [ ] I understand the system-wide impact of these changes on affected modules and performance.
- [ ] The design fits with the current website's architectural approaches.
- [ ] I have tested my changes extensively across browsers and screen sizes.
- [ ] Every line of code is personally understood and explainable.

### 📚 Resources & Support

**References:**
- [Next.js Documentation](https://nextjs.org/docs)
- [React Documentation](https://react.dev)
- [Tailwind CSS Documentation](https://tailwindcss.com/docs)
- [Vitest Documentation](https://vitest.dev)
- [Writing Blog Posts](https://github.com/hiero-ledger/hiero-website/blob/main/docs/blogs.md)

**🆘 Stuck?**
- [Community Calls](https://zoom-lfx.platform.linuxfoundation.org/meetings/hiero?view=week)
- [Discord](https://github.com/hiero-ledger/sdk-collaboration-hub/blob/main/guides/issue-progression/for-developers/discord.md)

### PR DESCRIPTION
**Pull request by aceppaluni**
# Description

This PR aims to add workflows to assist with triaging issues. The goal is to close off issues from interaction until they have been approved by a supporting team member

## Changes Made
- [x] Added... moderate-new-issues.yml and approved-issues.yml
- [ ] Modified...
- [ ] Fixed...

## Related Issues

Fixes: #446 

## Checklist
- [ ] Tests added/updated
- [ ] Documentation updated
- [ ] Linting passes
- [x] Branch up-to-date with main

## Deployment Notes

A link to testing: https://github.com/aceppaluni/hiero-website/issues/9

## Summary by CodeRabbit

* **Chores**
  * Implemented automated issue moderation workflow that labels new issues as pending review, posts status notifications, and applies temporary locks during the review process
  * Added automated approval workflow that removes pending status and unlocks approved issues with status comments

[Review Change Stack](https://app.coderabbit.ai/change-stack/hiero-ledger/hiero-website/pull/447?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: .github/workflows/moderate-new-issues.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          labels: pending-review
+
+      - name: Comment on issue
+        uses: peter-evans/create-or-update-comment@e8674b075228eee787fea43ef493e45ece1004c9 # v5.0.0
+        with:
+          issue-number: ${{ github.event.issue.number }}
+          body: |
+            Thanks for your submission.
+
+            This issue is awaiting approval from @hiero-website-committers and @hiero-website-maintainers.
```
[the remark this answers is not in the document]
Comment (danielmarv) [reply .github/workflows/moderate-new-issues.yml]: Thanks @aceppaluni 
but i think you can take this into consideration coz it is crucial

[the remark this answers is not in the document]
Comment (danielmarv) [reply .github/workflows/moderate-new-issues.yml]: you can consider this

**Review by danielmarv — DISMISSED** [review]

**Comment by danielmarv** [comment]
@aceppaluni 
Kindly update yout branch

File: .github/workflows/approve-issues.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    shell: bash
+
+permissions:
+  issues: write
+
+jobs:
+  approve:
+    if: github.event.label.name == 'approved'
+    name: Approve Issue
+    runs-on: ubuntu-latest
```
Comment (danielmarv) [inline .github/workflows/approve-issues.yml]:
```suggestion
    runs-on: hl-web-lin-md 
```

File: .github/workflows/moderate-new-issues.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  issues: write
+
+concurrency:
+  group: moderate-issues-${{ github.event.issue.number }}
+  cancel-in-progress: false
+
+jobs:
+  moderate:
+    name: Moderate Issue
+    runs-on: ubuntu-latest
```
Comment (danielmarv) [inline .github/workflows/moderate-new-issues.yml]:
```suggestion
    runs-on: hl-web-lin-md 
```

**Review by danielmarv — CHANGES_REQUESTED** [review]

**Review by danielmarv — DISMISSED** [review]

File: .github/workflows/moderate-new-issues.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      - name: Comment on issue
+        uses: peter-evans/create-or-update-comment@e8674b075228eee787fea43ef493e45ece1004c9 # v5.0.0
+        with:
+          issue-number: ${{ github.event.issue.number }}
+          body: |
+            Thanks for your submission.
+
+            This issue is awaiting approval from @hiero-ledger/hiero-website-committers and @hiero-ledger/hiero-website-maintainers.
+            
+      - name: Close issue temporarily
```
Comment (exploreriii) [inline .github/workflows/moderate-new-issues.yml]: if it is locked, must we close the issue?
if we close it, could we miss the issue?

File: .github/workflows/moderate-new-issues.yml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        uses: peter-evans/close-issue@05dff7ca07116e5707cfb3a3cd4af7bb5e43524c # v3.0.2
+        with:
+          issue-number: ${{ github.event.issue.number }}
+          comment: |
+            This issue is temporarily closed pending moderator approval.
+
```
Comment (exploreriii) [inline .github/workflows/moderate-new-issues.yml]: also while i understand we want to lock the issue, will we notice the issue is ready to be reviewed as it looks like any other issue on the outside?
maybe can add pending-triage label too

File: .github/workflows/moderate-new-issues.yml (hunk 4 above)
Comment (aceppaluni) [reply .github/workflows/moderate-new-issues.yml]: I can remove the close. This was added as precaution.

File: .github/workflows/moderate-new-issues.yml (hunk 5 above)
Comment (aceppaluni) [reply .github/workflows/moderate-new-issues.yml]: The bot adds a pending review label. However the name can be changed

Comment (exploreriii) [reply .github/workflows/moderate-new-issues.yml]: this works fine

**Review by exploreriii — DISMISSED** [review]

**Comment by aceppaluni** [comment]
@hiero-ledger/github-maintainers

**Review by danielmarv — DISMISSED** [review]

**Comment by aceppaluni** [comment]
@hiero-ledger/github-maintainers

**Comment by aceppaluni** [comment]
@hiero-ledger/github-maintainers

**Comment by aceppaluni** [comment]
@hiero-ledger/github-maintainers

**Comment by jwagantall** [comment]
@hiero-ledger/github-maintainers , please help us unblock this PR!

**Comment by jwagantall** [comment]
@aceppaluni , can we please rebase this change on top of main instead of merging main into the branch?

**Comment by aceppaluni** [comment]
Hi @jwagantall I have successfully rebased. I did have to merge main into the branch again as I did not see the branch update. Happy to rebase again if needed :)

File: .github/workflows/approve-issues.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        with:
+          labels: pending-review
+
+      - name: Unlock issue
+        run: |
+          gh api \
+            -X DELETE \
+            repos/${{ github.repository }}/issues/${{ github.event.issue.number }}/lock
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (andrewb1269) [inline .github/workflows/approve-issues.yml]:
```suggestion
            "repos/${REPO}/issues/${ISSUE_NUMBER}/lock"
        env:
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          REPO: ${{ github.repository }}
          ISSUE_NUMBER: ${{ github.event.issue.number }}
```

Add env vars so we are not open to a script injection.

File: .github/workflows/moderate-new-issues.yml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+            This issue is awaiting approval from @hiero-ledger/hiero-website-committers and @hiero-ledger/hiero-website-maintainers.
+            
+      - name: Lock issue (IMPORTANT FIX)
+        run: |
+          gh api \
+            -X PUT \
+            repos/${{ github.repository }}/issues/${{ github.event.issue.number }}/lock \
+            -f lock_reason=off-topic
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (andrewb1269) [inline .github/workflows/moderate-new-issues.yml]:
```suggestion
          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          REPO: ${{ github.repository }}
          ISSUE_NUMBER: ${{ github.event.issue.number }}
```

File: .github/workflows/moderate-new-issues.yml [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+          body: |
+            Thanks for your submission.
+
+            This issue is awaiting approval from @hiero-ledger/hiero-website-committers and @hiero-ledger/hiero-website-maintainers.
+            
+      - name: Lock issue (IMPORTANT FIX)
+        run: |
+          gh api \
+            -X PUT \
+            repos/${{ github.repository }}/issues/${{ github.event.issue.number }}/lock \
```
Comment (andrewb1269) [inline .github/workflows/moderate-new-issues.yml]:
```suggestion
            repos/${REPO}/issues/${ISSUE_NUMBER}/lock \
```

**Review by andrewb1269 — CHANGES_REQUESTED** [review]

**Comment by aceppaluni** [comment]
Closing and will open a fresh PR.
